lib/awilixManager.ts

```diff
@@ -132,6 +131,25 @@ export function getWithTags(diContainer: AwilixContainer, tags: string[]): Recor
   return resolvedComponents
 }
 
+export function getByPredicate(
+  diContainer: AwilixContainer,
+  predicate: (entity: any) => boolean,
+): Record<string, any> {
+  const enabledDependencies = Object.entries(diContainer.registrations).filter((entry) => {
+    return entry[1].enabled !== false
+  })
+
+  const resolvedComponents: Record<string, any> = {}
+  for (const entry of enabledDependencies) {
+    const resolvedElement = diContainer.resolve(entry[0])
+    if (predicate(resolvedElement)) {
+      resolvedComponents[entry[0]] = resolvedElement
+    }
+  }
```

Comment:
🟢 Nit, feel free to ignore.

`entry[0]` I suppose is the dependency key, could we make it a bit more readable with something like:
```
for (const [key] of enabledDependencies) {
...
}
```

Reply:
great idea, addressed!